colexec: minor fixes around the spilling queue

yuzefovich opened the pull request:
**colexecutils: fix and improve closing of the spilling queue**

Previously, we forgot to release the memory registered with the
allocator of the spilling queue in `Close`. That method is called by
`Reset`, so the previous behavior could result in the increased reported
(not actual) memory usage which is suboptimal. This commit fixes that
issue as well as makes minor improvements to release some references
eagerly.

Release note: None

**colexec: fix resetting of the hash aggregator in an edge case**

Previously, when resetting the hash aggregator, we would close the
spilling queue in case the input tuples tracking is needed. This is
incorrect (`Close` is assumed to be called only once the spilling queue
will no longer be used), so this commit switches to using `Reset`.

This bug, however, doesn't pose any real danger because currently we
would never reset the hash aggregator in case the input tuples tracking
is needed (which is the case only when the hash aggregator is used as
the in-memory operator of the disk spiller, and the in-memory operators
never get reset to be reused later). So this commit fixes more of
a philosophical bug and, thus, won't be backported (and doesn't have
a regression test).

Release note: None

yuzefovich committed:
colexecutils: fix and improve closing of the spilling queue

Previously, we forgot to release the memory registered with the
allocator of the spilling queue in `Close`. That method is called by
`Reset`, so the previous behavior could result in the increased reported
(not actual) memory usage which is suboptimal. This commit fixes that
issue as well as makes minor improvements to release some references
eagerly.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65818)
<!-- Reviewable:end -->

yuzefovich committed:
colexec: fix resetting of the hash aggregator in an edge case

Previously, when resetting the hash aggregator, we would close the
spilling queue in case the input tuples tracking is needed. This is
incorrect (`Close` is assumed to be called only once the spilling queue
will no longer be used), so this commit switches to using `Reset`.

This bug, however, doesn't pose any real danger because currently we
would never reset the hash aggregator in case the input tuples tracking
is needed (which is the case only when the hash aggregator is used as
the in-memory operator of the disk spiller, and the in-memory operators
never get reset to be reused later). So this commit fixes more of
a philosophical bug and, thus, won't be backported (and doesn't have
a regression test).

Release note: None

DrewKimball reviewed (approved):
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65818)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @michae2)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
